SecretManagement.LastPass.psm1

```diff
@@ -0,0 +1,114 @@
+    StayConnectedForceSwitchMissing = 'StayConnected is a sensitive operation that save the LastPass decryption key on your hard drive. If you want to proceed, reissue the command specifying the force parameter.'
+    Unregister_NotLpass_1           = "The specified vault is not a $ModuleName vault (VaultType: {0}"
+}
+
+function Connect-LastPass {
+    [CmdletBinding()]
+    param (
+        [String]$Vault,
+        [String]$User,
+        [Switch]$Trust,
+        [Switch]$StayConnected,
+        [Switch]$Force 
+    )
+    $Arguments = [System.Collections.Generic.List[String]]@('login')
+    if ($trust) { $Arguments.Add('--trust') }
+    if ($StayConnected) { 
+        if (! $Force) { Throw [System.Management.Automation.PSArgumentException] $ErrorMessages.StayConnectedForceSwitchMissing }
+        $Arguments.Add('--plaintext-key') 
+        $Arguments.Add('--force')
+    }
+    $Arguments.Add($User)
+    $VaultParams = Get-VaultParams -Vault $Vault
+    Invoke-lpass -Arguments $Arguments -VaultParams $VaultParams
+}
+
+function Disconnect-LastPass {
+    [CmdletBinding()]
+    param (
+        [String]$Vault
+    )
+    $Arguments = [System.Collections.Generic.List[String]]@('logout', '--force')
+    $VaultParams = Get-VaultParams -Vault $Vault
+    Invoke-lpass -Arguments $Arguments -VaultParams $VaultParams
+}
+
+function Register-LastPassVault {
+    [CmdletBinding()]
+    param (
+        [String]$Vault,
+        [switch]$wsl,
+        [Switch]$Detailed,
+        [String]$Path
+    )
+
+    $Params = @{
+        ModuleName      = 'SecretManagement.LastPass'
+        Name            = if ('' -ne $Vault) {$Vault} else {$ModuleName}
+        VaultParameters = @{}
+    }
+    if ($wsl -eq $true) { $Params.VaultParameters.Add('wsl', $true) }
+    if ($Path -ne '') { $Params.VaultParameters.Add('lpassPath', $Path) }
+    if ($Detailed -eq $true) { $Params.VaultParameters.Add('outputType','Detailed') }
+    if ($VerbosePreference -eq 'Continue') {$Params.add('Verbose',$true)}
+
+    Register-SecretVault @Params
+}
+function Unregister-LastPassVault {
+    [CmdletBinding()]
+    param (
+        [String]$Vault
+    )
+    $Params = @{Name = if ('' -ne $Vault) { $Vault } else { $ModuleName } }
+    if ($VerbosePreference -eq 'Continue') {$Params.Add('Verbose',$true)}
+    
+    $Vault = Get-SecretVault -Name $params.Name -ErrorAction Stop
+    if ($Vault.ModuleName -ne $ModuleName) { Throw $ErrorMessages.Unregister_NotLpass_1 -f $Vault.ModuleName}
+    Unregister-SecretVault @Params 
+}
+
+function Sync-LastPassVault {
+    [CmdletBinding()]
+    param ([String]$Vault)
+    $VaultParams = Get-VaultParams -Vault $Vault
+    Invoke-lpass -Arguments 'sync' -VaultParams $VaultParams
+}
+Function Get-VaultParams {
+    Param($Vault)
+    if ([String]::IsNullOrEmpty($Vault)) { 
+        $AllVaults = Get-SecretVault | Where-Object ModuleName -eq $ModuleName
```

Comment:
wouldn't it be best to use `-Name` on `Get-SecretVault` and then handle when there isn't one called `$ModuleName`?

Reply:
[Done](https://github.com/TylerLeonhardt/SecretManagement.LastPass/pull/18/commits/29e86919b71bd42ef6501ed722ba3764d3ae3457)

Reply:
do we need the switch anymore? Can you register the same vault name twice?

Reply:
@TylerLeonhardt 
You cannot register the same vault twice. 
The behavior is this.

- If user do not provide a vault name (`[String]::IsNullOrEmpty($Vault)`) 
- We get the vault called SecretManagement.LastPass and return its parameter
BUT

If there's no vault with such name, the switch statement kick in and return the vault parameter of any `SecretManagement.LastPass` vault under the condition that there is only 1. 

If there's a different number, an error is thrown to ask the user to provide a vault name (multiple) or to indicate there's no lastpass vault at all. 

Get-VaultParams is actually not used at all by Register / Unregister

Reply:
ahhh I see. That makes sense! I didn't see the _ModuleName_ in the where clause and thought you had specified Name. Maybe worth a comment.